guides/routing/htmlforms.html

```diff
@@ -0,0 +1,246 @@
+    <p class="sentence">It's good practice to keep value types (in this case models) and logic separated. Create a new file, to contain the models, called Models.swift as follows:</p>
+    <ul>
+      <li>1. In Xcode right click on the Application folder and select New File.</li>
+      <li>2. Select Swift File in the UI pop up and click next.</li>
+      <li>3. In the Save As field, name the file Models.swift.</li>
+      <li>4. Ensure Group is Application and target is Application.</li>
+      <li>5. Click Create to create the file.</li>
+    </ul>
+    <p class="sentence">For now, let's create a very simple model:</p>
+    <p class="sentence">1. Open Models.swift</p>
+    <p class="sentence">2. Copy in the following code:</p>
+    <pre><code class="language-swift">import KituraContracts
+
+struct Form: Codable, QueryParams {
+    let name: String
+    let age: Int?
+    let isDeveloper: Bool
+}</code></pre>
+    <p class="sentence">This provides a simple representation of a data we will recieve from our form. We marked it as Codable so it can be used in Codable routes and as QueryParams so it can be decoded from query parameters on a GET request.</p>
+    <p class="sentence">If the forms "age" field is empty, it will send the empty string in the query and Kitura will decode the value to nil. If the forms "isDeveloper" field is empty, it will send the empty string in the query and Kitura will decode the value to false.
+    <p class="sentence">Now we have a model we can use, we can create our routes!</p>
+
+
+    <h2><a name="codable-routes" style="padding-top: 100px;"></a>Setup the Codable Routes</h2>
+    <p class="sentence">We will now create a <a href="https://www.kitura.io/guides/routing/codablerouting.html target="_blank"">Codable route</a> to recieve the POST request from an HTML form.</p>
+    <p class="sentence">1. Open Application.swift</p>
+    <p class="sentence">2. Inside the "postInit" function, Register the route handler:</p>
+    <pre><code class="language-swift">router.post("/codable", handler: postFormHandler)</code></pre>
+    <p class="sentence">3. After the "postInit" function, add your "postFormHandler" function:
+    <pre><code class="language-swift">func postFormHandler(user: Form, respondWith: (Form?, RequestError?) -> Void) {
+    print("Codable POST route: \(user.name)")
+    if let age = user.age { print("is \(age) years old") }
+    if(user.isDeveloper) { print("and they are a developer") }
+    respondWith(user, nil)
+}</code></pre>
+    <p class="sentence">4. Restart your server and go to <a href="http://localhost:8080/formwebpage.html" target="_blank">localhost:8080/formwebpage.html</a>. Send the "Post to Codable route" form and in the Xcode logger you should see our printed message with the form data.</p>
+
+    <p class="sentence">We will now create a <a href="https://www.kitura.io/guides/routing/codablerouting.html target="_blank"">Codable route</a> to recieve the GET request from an HTML form.</p>
```

Comment:
This link is broken as above but I suggest removing the link here as it's already given above.